sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/StorageSpec.groovy

```diff
@@ -32,6 +34,12 @@ class StorageSpec extends Specification {
     private InterceptorManager interceptorManager
     private StorageResourceNamer namer
 
+    def setupSpec() {
+        Hooks.onErrorDropped({throwable -> LOGGER.logThrowableAsError(throwable) })
+        Schedulers.onHandleError({ thread, throwable -> LOGGER.logThrowableAsError(throwable) })
+        Thread.setDefaultUncaughtExceptionHandler({ thread, throwable -> LOGGER.logThrowableAsError(throwable) })
```

Comment:
Should we put that in some static initializer ? `AbstractGlobalExtension`/`IGlobalExtension` from spock has `start()` hook. This might be good place to put it. See https://github.com/Azure/azure-sdk-for-java/pull/22573 for reference how to create global extension.

We should also erradicate things like this https://github.com/Azure/azure-sdk-for-java/search?q=%22Hooks.onErrorDropped%22 scattered across the test codebase.

Reply:
If we choose to actually merge this change I think those would be a better location for this